server/models/server-service.js

```diff
@@ -249,4 +253,55 @@ module.exports = function extendServerService(ServerService) {
     });
   }
   ServerService.prototype._callOnInstances = _callOnInstances;
+
+  ServerService.__changeEmitter = new EventEmitter();
+  function createChangeStream(res, cb) {
+    if (typeof res === 'function') {
+      cb = res;
+      res = undefined;
+    }
+
+    var changes = new PassThrough({objectMode: true});
+    changes._serviceId = this.id;
+    debug('change stream %s created', changes._serviceId);
+    ServerService.__changeEmitter.on(
+      'svc-' + changes._serviceId,
+      publishChange
+    );
+
+    changes.destroy = function() {
+      debug('change stream %s destroyed', changes._serviceId);
+      changes.removeAllListeners('error');
```

Comment:
Should this be removing all listeners instead of just these 2? Seems to me like it could theoretically still be emitting `data` events to someone holding on to a ref.. but if it tried to emit an error as a result, nothing would be listening so it would crash the app.